lib/sycamore/sycamore/data/table.py

```diff
@@ -0,0 +1,187 @@
+    # cells spanning multiple rows/columns.
+    #
+    # One of the main decisions here is how to render cells that span multiple rows and columns,
+    # since they are not nativey supported in CSV. The standard approaches are (1) to duplicate
+    # the cell value in each row/column or (2) to put the value in the first row/column in the cell
+    # and empty values in the other. TableTransformers do the former, The textractor library supports
+    # both, but defaults to the latter. Since our primary use case is preparing context for LLMs,
+    # we speculate that duplication may create confusion, so we default to only displaying a cells
+    # content for the first row/column for which it is applicable. The exception is for header rows,
+    # where we duplicate values to each columnn to ensure that every column has a fully qualified header.
+    def to_pandas(self) -> DataFrame:
+        # Find all row nums containing cells marked as headers.
+        header_rows = sorted(set((row_num for cell in self.cells for row_num in cell.rows if cell.is_header)))
+
+        # Find the number of initial rows that are header rows. This currently treats rows that are
+        # marked as headers but are not part of the prefix of rows in the table as regular (non-header)
+        # rows. This is wrong in general, but necessary for conversion to simpler formats like CSV.
+        # Note that we also don't handle the case where only part of a row is a header. If any cell in
+        # the row is marked as a header, all cells are treated as such for this conversion.
+        i = -1
+        for i in range(len(header_rows)):
+            if header_rows[i] != i:
+                break
+
+        max_header_prefix_row = i
+
+        table_array = np.empty([self.num_rows, self.num_cols], dtype="object")
+        if len(self.cells) > 0:
+            for cell in self.cells:
+                # We treat header cells that are not at the beginning of the table
+                # as regular cells.
+                if cell.is_header and cell.rows[0] <= max_header_prefix_row:
+                    # Put the value in the first row and all the columns.
+                    for col in cell.cols:
+                        table_array[cell.rows[0], col] = cell.content
+
+                    # Put an empty string in each of the other rows. We want the
+                    # header included in every column, but we don't need it repeated
+                    # multiple times for a single column.
+                    for row in cell.rows[1:]:
+                        for col in cell.cols:
+                            table_array[row, col] = ""
+
+                else:
+                    table_array[cell.rows[0], cell.cols[0]] = cell.content
+                    for row in cell.rows[1:]:
+                        for col in cell.cols[1:]:
+                            table_array[row, col] = ""
+
+        header = table_array[: max_header_prefix_row + 1, :]
```

Comment:
does this handle for cases that left most column is also header, or do we want such case handled?

Reply:
Interesting. I did not handle this case. It looks like Pandas does support this, though I guess it won't impact the CSV representation. Happy to play around with it, though not sure it will matter in any of our initial use cases.